core/src/test/java/org/apache/calcite/test/RexProgramBuilderBase.java

```diff
@@ -496,7 +496,7 @@ protected RexNode vInt() {
    */
   protected RexNode vInt(int arg) {
     assertArgValue(arg);
-    return rexBuilder.makeFieldAccess(getDynamicParam(nonNullableInt, "int"), arg);
+    return rexBuilder.makeFieldAccess(getDynamicParam(nullableInt, "int"), arg);
```

Comment:
why is this change?

Reply:
@vlsi , it seemed like a typo. In the javadoc, it says:
```
  /**
   * Creates {@code nullable int variable} with index of {@code arg} (0-based).
   * The resulting node would look like {@code ?0.int3} if {@code arg} is {@code 3}.
   *
   * @return nullable int variable with given index (0-based)
   */
```
In addition, there is a ```protected RexNode vIntNotNull(int arg)``` just below.

Reply:
The method was clearly not doing what its description was saying.

Reply:
Why do you think so?

Reply:
Please check

```
  // Note: JUnit 4 creates new instance for each test method,
  // so we initialize these structures on demand
  // It maps non-nullable type to struct of (10 nullable, 10 non-nullable) fields
  private Map<RelDataType, RexDynamicParam> dynamicParams;
```

Reply:
JUnit 4 does incur high overhead since it re-creates test state for each test method.
It is truly expensive to create individual structs for each test method, so test code is using the same RexDynamicParam instance for both nullable and non-nullable fields.

@jcamachor , `vIntNotNull` uses `arg + MAX_FIELDS` while `vInt` uses just `arg` index.

Clearly one should not change code at random while adding zero tests

Reply:
Agreed, we can create a follow-up either to fix the code or to fix the comment. Probably @kgyrtkirk  has more context.

Reply:
I've noticed it at some point; looked like a typo - and I was worried that all the tests sofar have run mostly with with non nullable booleans...
I'll remove this change from this patch.